### PR TITLE
Fix a weird C++/V8 compatibility bug

### PR DESCRIPTION
**Pull request by pksunkara**
I don't even know why the tests in #79 were failing. All of them were passing locally but failing on CI servers. I had to use CircleCI SSH feature and did trial and error experiments to get a working version. I still don't know what exactly was causing it in the first place.

This PR contains the following:

* Code that fixes the above mentioned weird issue.
* Adds a new test for checking API Refract.
* Removes the MSON Expansion from existing tests.

### CODE REVIEW

**Comment by pksunkara** [comment]
Need to merge apiaryio/drafter#88 first so that I can make another PR on drafter which needs to be merged before this.

**Comment by pksunkara** [comment]
This is now ready to be merged.

**Comment by zdne** [comment]
> I don't even know why the tests in #79 were failing. All of them were passing locally but failing on CI servers. I had to use CircleCI SSH feature and did trial and error experiments to get a working version. I still don't know what exactly was causing it in the first place.

Is this still the case or it was caused by the missing includes?

**Comment by pksunkara** [comment]
> Is this still the case or it was caused by the missing includes?

Yes, it is still the case. I still don't why the tests were failing on all CI services. The `iterator` headers only fix the issues raised by Visual Studio which is completely different issue than the above one.

File: src/options_parser.cc [hunk 1]
```diff
@@ -42,14 +42,13 @@ OptionsResult* protagonist::ParseOptionsObject(Handle<Object> optionsObject) {
         }
         else if (TypeOptionKey == *String::Utf8Value(key)) {
             // TypeOption
-            const char *strValue = *String::Utf8Value(value);
```
Comment (zdne) [inline src/options_parser.cc]: FYI This should have been

```c++
String::Utf8Value strValue(value);
```

then you could use it as 

```c++
if (TypeOptionAst == *strValue) {
```

Comment (zdne) [inline src/options_parser.cc]: It is actually important to understand why this was the issue. Basically you were relying on a compiler's implementation. Assigning a temporary value to `const char *strValue` which, in some implementation didn't have to be there when you were dereferencing the `strValue` .

Comment (pksunkara) [inline src/options_parser.cc]: Tried that first before anything. Compilation was failing on circleCI.

Comment (zdne) [inline src/options_parser.cc]: @pksunkara is it clear?

**Comment by pksunkara** [comment]
Updated. Now this PR optimises the code in `options_parser.cc` a bit too.

**Comment by zdne** [comment]
@pksunkara great job! waiting for CI

**Pull request merged**